### PR TITLE
chore: enhanced sidebar and navigation

### PR DESCRIPTION
**Pull request by themr0c**
* Updated default sidebar. https://docusaurus.io/docs/sidebar#auto-collapse-sidebar-categories
* Hideable sidebar. See https://docusaurus.io/docs/sidebar#hideable-sidebar
* Auto-collapse sidebar categories. See https://docusaurus.io/docs/sidebar#auto-collapse-sidebar-categories
* Navbar linked to the sidebar. See  https://docusaurus.io/docs/sidebar#auto-collapse-sidebar-categories
* references: https://github.com/containers/podman-desktop/issues/1294

Better UX, but the sidebar is still hidden on smaller screens.
    
Signed-off-by: Fabrice Flore-Thébault <ffloreth@redhat.com>

### CODE REVIEW

File: website/sidebars.js [hunk 1]
```diff
@@ -1,31 +1,8 @@
...
-/** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */
-const sidebars = {
-  // By default, Docusaurus generates a sidebar from the docs folder structure
-  tutorialSidebar: [{ type: 'autogenerated', dirName: '.' }],
-
-  // But you can create a sidebar manually
-  /*
-  tutorialSidebar: [
+module.exports = {
+  mySidebar: [
```
Comment (benoitf) [inline website/sidebars.js:6]: it looks like all these changes on this file are like a no-op change 
You also removed the TypeScript types / checks

Comment (themr0c) [reply website/sidebars.js:6]: It sticks to Docusaurus documentation.

Comment (benoitf) [reply website/sidebars.js:6]: the types should not be removed

```
// @ts-check

/** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */

```

also from what I see you didn't introduced any changes in this file except removing all the comments

File: website/sidebars.js [hunk 2]
```diff
@@ -1,31 +1,8 @@
...
-
-/** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */
-const sidebars = {
-  // By default, Docusaurus generates a sidebar from the docs folder structure
-  tutorialSidebar: [{ type: 'autogenerated', dirName: '.' }],
-
-  // But you can create a sidebar manually
-  /*
-  tutorialSidebar: [
+module.exports = {
```
Comment (themr0c) [inline website/sidebars.js]:
```suggestion
// @ts-check

/** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */

module.exports = {
```

File: website/sidebars.js (hunk 1 above)
Comment (themr0c) [reply website/sidebars.js:6]: The code looked to me as boilerplate from an older version.
I took the code from the current docs, for clarity: it's exactly the same as in the docs, so there is no reason to search to understand the differences.

Comment (benoitf) [reply website/sidebars.js:6]: the code is generated from the docusaurus generator and it includes the Types so we have completion with IDE
here you removed all type checking entries

**Review by benoitf — DISMISSED** [review]
please keep the typescript checks/types

File: website/sidebars.js (hunk 1 above)
Comment (themr0c) [reply website/sidebars.js:6]: // @ts-check

/** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */

are back, what else is missing?

Comment (benoitf) [reply website/sidebars.js:6]: they need to be where the field/type is used

File: website/sidebars.js [hunk 3]
```diff
@@ -1,31 +1,12 @@
...
+  mySidebar: [
     {
-      type: 'category',
-      label: 'Tutorial',
-      items: ['hello'],
+      type: 'autogenerated',
+      dirName: '.', // generate sidebar from the docs folder (or versioned_docs/<version>)
```
Comment (benoitf) [inline website/sidebars.js]: Full code should be


```js
// @ts-check

/** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */
const sidebars = {
  // // generate sidebar from the docs folder (or versioned_docs/<version>)
  mySidebar: [{ type: 'autogenerated', dirName: '.' }],

};

module.exports = sidebars;
```

File: website/sidebars.js (hunk 1 above)
Comment (themr0c) [reply website/sidebars.js:6]: Last commit should be fine.

**Review by cdrage — APPROVED** [review]
LGTM!

Tested and works well / LGTM.

**Review by cdrage — APPROVED** [review]

**Pull request merged**